### YamlMime:ManagedReference
items:
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs
  id: DataGridViewDataErrorEventArgs
  children:
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  langs:
  - csharp
  name: DataGridViewDataErrorEventArgs
  nameWithType: DataGridViewDataErrorEventArgs
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs
  type: Class
  summary: "Udostępnia dane dla <xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref> zdarzeń."
  remarks: "Obsługa <xref:System.Windows.Forms.DataGridView.DataError>zdarzenie pozwala obsługi wyjątków zgłaszanych przez kod poza formantu (na przykład przez zewnętrznego źródła danych).</xref:System.Windows.Forms.DataGridView.DataError> Użyj <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A>Właściwości w celu określenia stanu <xref:System.Windows.Forms.DataGridView>w czasie wystąpienia wyjątku.</xref:System.Windows.Forms.DataGridView> </xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A> Użyj <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A>właściwość, aby pobrać Dane wyjątku.</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> Do obsługi wyjątku przez programy obsługi zdarzeń dodatkowe, należy ustawić <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>właściwości `true`.</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>       <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>i <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A>Właściwości zwykle wskazują komórki wystąpił błąd danych, w którym.</xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A> </xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> Po wystąpieniu błędu w zewnętrznym źródłem danych, jednak źródła danych może nie zawiera kolumny, w którym wystąpił błąd. W takim przypadku <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>Właściwość zwykle wskazuje kolumny bieżącej komórki w chwili wystąpienia błędu.</xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>"
  example:
  - "The following code example demonstrates how to respond to information provided by the DataGridViewDataErrorEventArgs class. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/t-system.windows.forms.d_56_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/t-system.windows.forms.d_56_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/t-system.windows.forms.d_56_1.vb)]"
  syntax:
    content: 'public class DataGridViewDataErrorEventArgs : System.Windows.Forms.DataGridViewCellCancelEventArgs'
  inheritance:
  - System.ComponentModel.CancelEventArgs
  - System.Windows.Forms.DataGridViewCellCancelEventArgs
  implements: []
  inheritedMembers:
  - System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex
  - System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  id: '#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)'
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  type: Constructor
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "Inicjuje nowe wystąpienie klasy <xref href=&quot;System.Windows.Forms.DataGridViewDataErrorEventArgs&quot;> </xref> klasy."
  syntax:
    content: public DataGridViewDataErrorEventArgs (Exception exception, int columnIndex, int rowIndex, System.Windows.Forms.DataGridViewDataErrorContexts context);
    parameters:
    - id: exception
      type: System.Exception
      description: "Wyjątek, który wystąpił."
    - id: columnIndex
      type: System.Int32
      description: "Indeks kolumny komórki, która wywołała <xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref>."
    - id: rowIndex
      type: System.Int32
      description: "Indeks wiersza komórki, która wywołała <xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref>."
    - id: context
      type: System.Windows.Forms.DataGridViewDataErrorContexts
      description: "Bitowe połączenie <xref href=&quot;System.Windows.Forms.DataGridViewDataErrorContexts&quot;> </xref> wartości, wskazując kontekst, w którym wystąpił błąd."
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  id: Context
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "Pobiera szczegóły dotyczące stanu <xref href=&quot;System.Windows.Forms.DataGridView&quot;> </xref> wystąpienia błędu."
  remarks: "Elementy członkowskie <xref:System.Windows.Forms.DataGridViewDataErrorContexts>wyliczenia mogą być łączone przy użyciu operatora testu koniunkcji `OR` operator do reprezentowania stanu z danymi <xref:System.Windows.Forms.DataGridView>wystąpienia błędu danych.</xref:System.Windows.Forms.DataGridView> </xref:System.Windows.Forms.DataGridViewDataErrorContexts> Na przykład, jeśli użytkownik wprowadzi wartość komórki nieprawidłowe (na przykład w komórce, która wymaga datę, wprowadzając nazwę), a następnie wybiera inną komórkę <xref:System.Windows.Forms.DataGridView>spróbuje przekazać wartość komórki nieprawidłowy.</xref:System.Windows.Forms.DataGridView> Jeśli zatwierdzanie nie powiodło się, <xref:System.Windows.Forms.DataGridView>zgłosi <xref:System.Windows.Forms.DataGridView.DataError>zdarzeń, w których właściwość kontekstu będzie miał wartość i <xref:System.Windows.Forms.DataGridViewDataErrorContexts> <xref:System.Windows.Forms.DataGridViewDataErrorContexts>.</xref:System.Windows.Forms.DataGridViewDataErrorContexts> </xref:System.Windows.Forms.DataGridViewDataErrorContexts> </xref:System.Windows.Forms.DataGridView.DataError> </xref:System.Windows.Forms.DataGridView>"
  example:
  - "The following code example demonstrates how to investigate the error context. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_134_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_134_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_134_1.vb)]"
  syntax:
    content: public System.Windows.Forms.DataGridViewDataErrorContexts Context { get; }
    return:
      type: System.Windows.Forms.DataGridViewDataErrorContexts
      description: "Bitowe połączenie <xref href=&quot;System.Windows.Forms.DataGridViewDataErrorContexts&quot;> </xref> wartości, które określa kontekst, w którym wystąpił błąd."
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  id: Exception
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "Pobiera wyjątek, który reprezentuje błąd."
  remarks: "Do obsługi wyjątku przez więcej niż jedną <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>, zgłosić wyjątek ponownie po zakończeniu obsługi przez ustawienie <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>właściwości `true`.</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>"
  example:
  - "The following code example demonstrates checking the Exception property. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_249_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_249_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_249_1.vb)]"
  syntax:
    content: public Exception Exception { get; }
    return:
      type: System.Exception
      description: "<xref:System.Exception>Reprezentujący błąd.</xref:System.Exception>"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  id: ThrowException
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "Pobiera lub ustawia wartość wskazującą, czy ma zostać zgłoszony wyjątek po <xref href=&quot;System.Windows.Forms.DataGridViewDataErrorEventHandler&quot;> </xref> zakończeniu delegata działania z nim."
  remarks: "Ustaw tą właściwość na `false` Jeśli <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>prowadzonych wyjątek reprezentowany przez <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A>Właściwości, a nie chcesz propagowanie wyjątku do innego <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>która może obsłużyć <xref:System.Windows.Forms.DataGridView.DataError>zdarzeń.</xref:System.Windows.Forms.DataGridView.DataError> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> </xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>       Na potrzeby debugowania, może być przydatne do propagowanie wyjątku i przejrzyj ślad stosu, aby uzyskać informacje dotyczące kontekstu błędu."
  example:
  - "The following code example demonstrates using the ThrowException property to indicate that the exception should not be thrown. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_202_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_202_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_202_1.vb)]"
  syntax:
    content: public bool ThrowException { get; set; }
    return:
      type: System.Boolean
      description: "<xref uid=&quot;langword_csharp_true&quot; name=&quot;true&quot; href=&quot;&quot;></xref>Jeśli wyjątek powinien zostać zgłoszony; w przeciwnym razie <xref uid=&quot;langword_csharp_false&quot; name=&quot;false&quot; href=&quot;&quot;> </xref>. Wartość domyślna to <xref uid=&quot;langword_csharp_false&quot; name=&quot;false&quot; href=&quot;&quot;> </xref>."
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*
  exceptions:
  - type: System.ArgumentException
    commentId: T:System.ArgumentException
    description: "Gdy ustawienie dla tej właściwości <xref uid=&quot;langword_csharp_true&quot; name=&quot;true&quot; href=&quot;&quot;> </xref>, <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*>wartość właściwości jest <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>.</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*>"
  platform:
  - net462
references:
- uid: System.Windows.Forms.DataGridViewCellCancelEventArgs
  isExternal: false
  name: System.Windows.Forms.DataGridViewCellCancelEventArgs
- uid: System.ArgumentException
  isExternal: true
  name: System.ArgumentException
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Int32
  parent: System
  isExternal: true
  name: Int32
  nameWithType: Int32
  fullName: System.Int32
- uid: System.Windows.Forms.DataGridViewDataErrorContexts
  parent: System.Windows.Forms
  isExternal: false
  name: DataGridViewDataErrorContexts
  nameWithType: DataGridViewDataErrorContexts
  fullName: System.Windows.Forms.DataGridViewDataErrorContexts
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
- uid: System.Boolean
  parent: System
  isExternal: true
  name: Boolean
  nameWithType: Boolean
  fullName: System.Boolean
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: DataGridViewDataErrorEventArgs
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
